test/functional/ui/screen_basic_spec.lua

```diff
@@ -73,33 +73,29 @@ describe('Screen', function()
   describe(':suspend', function()
     it('is forwarded to the UI', function()
       local function check()
-        if not screen.suspended then
-          return 'Screen was not suspended'
-        end
+        eq(true, screen.suspended)
       end
```

Comment:
That happens _precisely_ because `eq(false, screen.suspended)` here _is_ a failure-after-success condition, you could get either success, failure, or failure-after-success warning in different runs. Change it to `eq('test', screen.suspended)` and you get a guaranteed failure.

I've induced many correct test failures both in `basic_spec` and the cursor shape tests, this wouldn't be possible if `pcall()` didn't catch test failures. Also, how would test failures possibly stop the test execution if not by raising lua error?